src/components/course/routes/ExternalCourseEnrollment.jsx

```diff
@@ -77,7 +77,7 @@ const ExternalCourseEnrollment = () => {
   //   current date is outside the enrollment window of the run.
   if (userSubsidyApplicableToCourse.subsidyType === LEARNER_CREDIT_SUBSIDY_TYPE) {
     const canRedeemDataCourseRun = redeemabilityPerContentKey.find(r => r.contentKey === courseRunKey);
-    if (!canRedeemDataCourseRun?.canRedeem) {
+    if (!canRedeemDataCourseRun?.canRedeem && !canRedeemDataCourseRun?.hasSuccessfulRedemption) {
```

Comment:
Upon successful redemption of an Exec Ed course, the API calls to `can-redeem`, `credits_available`, and `enterprise_course_enrollments` are invalidated/re-fetched in parallel.

When the `can-redeem` API resolves first, it's response will update to have `canRedeem: false` and `hasSuccessfulRedemption: true`; however, this code path would handle this case as showing the `<NotFoundPage />` component, which isn't correct.

Once the other API calls resolve, the user is navigated to the `complete` route for the enrollment confirmation screen as intended.

This fix ensures we don't show the `<NotFoundPage />` while resolving the other API calls.